src/irr/asset/CGraphicsPipelineLoaderMTL.cpp

```diff
@@ -328,6 +780,168 @@ const char* CGraphicsPipelineLoaderMTL::readTexture(const char* _bufPtr, const c
     return _bufPtr;
 }
 
+std::pair<core::smart_refctd_ptr<ICPUSpecializedShader>, core::smart_refctd_ptr<ICPUSpecializedShader>> CGraphicsPipelineLoaderMTL::getShaders(bool _hasUV)
+{
+    auto vs = getDefaultAsset<ICPUSpecializedShader, IAsset::ET_SPECIALIZED_SHADER>(_hasUV ? VERT_SHADER_UV_CACHE_KEY : VERT_SHADER_NO_UV_CACHE_KEY, m_assetMgr);
+    auto fs = getDefaultAsset<ICPUSpecializedShader, IAsset::ET_SPECIALIZED_SHADER>(_hasUV ? FRAG_SHADER_UV_CACHE_KEY : FRAG_SHADER_NO_UV_CACHE_KEY, m_assetMgr);
+
+    return { std::move(vs), std::move(fs) };
+}
+
+auto CGraphicsPipelineLoaderMTL::loadImages(const char* _relDir, const CMTLPipelineMetadata::SMtl& _mtl, SContext& _ctx) -> image_views_set_t
+{
+    images_set_t images;
+
+    std::string relDir = _relDir;
+    for (uint32_t i = 0u; i < images.size(); ++i)
+    {
+        SAssetLoadParams lp;
+        if (_mtl.maps[i].size())
+        {
+            auto bundle = interm_getAssetInHierarchy(m_assetMgr, relDir+_mtl.maps[i], lp, _ctx.topHierarchyLevel+ICPURenderpassIndependentPipeline::IMAGE_HIERARCHYLEVELS_BELOW);
```

Comment:
you're not using the override?

Reply:
no, why would i? for what?

Reply:
override always needs to be used (more accurately, passed through) when loaders call other loaders (don't call `interm_getAssetInHierarchy` directly) so the user/developer can intervene in the loading of textures and dependent resources.

Reply:
override was never used directly in loader to load assets. Thats what 1interm_getAssetInHierarchy` is for. Then loader override is used by `IAssetManager::getAssetInHierarchy()`. However u're right i should be using overload of `interm_getAssetInHierarchy` that let u pass loader override ptr (fixed in next commit)

Reply:
yeah I mean use the function and pass `override` to it